src/run_releasenotes_check.py

```diff
@@ -0,0 +1,94 @@
+import logging
+import os
+import subprocess
+from typing import Any, List
+
+from pytablewriter import MarkdownTableWriter
+
+from checkout_workflow.checkout_args import CheckoutArgs
+from git.git_repository import GitRepository
+from manifests.input_manifest import InputComponentFromSource, InputManifest
+from system import console
+from system.temporary_directory import TemporaryDirectory
+
+# in format yyyy-mm-dd, example 2022-07-26
+gitLogDate = os.getenv('GIT_LOG_DATE')
+# Github issue to add the generated markdown table as a new comment
+gitIssueNumber = os.getenv('GIT_ISSUE_NUMBER')
+# Boolen if set to null, just prints on the console.
+addComment = os.getenv('ADD_COMMENT', default="true")
+# Token used to add comment on an issue
+userToken = os.getenv('GITHUB_TOKEN')
+
+
+def main(gitLogDate: str, addComment: str) -> int:
+    value_matrix = []  # type: List[Any]
+    args = CheckoutArgs()
+    console.configure(level=args.logging_level)
+    manifest = InputManifest.from_file(args.manifest)
+    writer = MarkdownTableWriter(
+        table_name=f"{manifest.build.name} Commits(after {gitLogDate}) & Release Notes info",
+        headers=["Repo", "Branch", "CommitID", "Release Notes"],
+        value_matrix=value_matrix
+    )
+    with TemporaryDirectory(keep=True, chdir=True) as work_dir:
+        logging.info(f"Checking out into {work_dir.name}")
+        for component in manifest.components.select():
+            logging.info(f"Checking out {component.name}")
+            repo_list = []  # type: List[Any]
+            if type(component) is InputComponentFromSource:
+                with GitRepository(
+                    component.repository,
+                    component.ref,
+                    os.path.join(work_dir.name, component.name),
+                    component.working_directory,
+                ) as repo:
+                    logging.debug(f"Checked out {component.name} into {repo.dir}")
+                    gitLogCmd = f'git -C {repo.dir} log --after={gitLogDate} --pretty=format:"%h"'
+                    gitLog = subprocess.check_output(gitLogCmd.split()).decode()
```

Comment:
Instead of using git version 2+ and `-C` to define working dir, it is better to use the `cwd` param in `subprocess` to switch current working dir.

Reply:
Also please verify that the `split()` is happening on output, not on command.

Reply:
That is required to make sure `subprocess` takes the command as follows
```['git', 'log', '--after=2022-07-26', '--pretty=format:"%h"']```

`cwd` is good idea to go then `-C` can be eliminated, which is available in 2.x version of git, pushed the change in my latest commit.
Thank you

Reply:
Thanks.